utils/utils.go

```diff
@@ -757,3 +772,44 @@ func getHomeDir() (string, error) {
 func IsDockerCmd(kind CmdType) bool {
 	return (kind == DockerRun || kind == DockerStart || kind == DockerCompose)
 }
+
+func CheckAuth(ctx context.Context, host, token string, hardReset bool, logger *zap.Logger) (string, bool, string, string, error) {
```

Comment:
I think these login related structs and methods below should go into a separate package in platform and ideally even interfaced.